tools/ts/build_lib.ts

```diff
@@ -0,0 +1,301 @@
+import { writeFileSync } from "fs";
+import { join, resolve } from "path";
+import * as prettier from "prettier";
+import Project, {
+  ExpressionStatement,
+  ModuleKind,
+  ModuleResolutionKind,
+  NamespaceDeclarationKind,
+  ScriptTarget,
+  SourceFile,
+  Type,
+  TypeGuards
+} from "ts-simple-ast";
+import {
+  addInterfaceProperty,
+  addNamespaceDeclaration,
+  addSourceComment,
+  addVariableDeclaration,
+  INTERNAL_GENERATED_SPECIFIER,
+  loadFiles,
+  loadDtsFiles,
+  setRootPath,
+  transferImportDeclarations
+} from "./ast_util";
+
+// this is very simplistic argument parsing, just enough to integrate into
+// the build scripts, versus being very robust
+let basePath = process.cwd();
+let buildPath = join(basePath, "out", "debug");
+let outFile = join(buildPath, "gen", "lib", "lib.d.ts");
+let silent = false;
+
+process.argv.forEach((arg, i, argv) => {
+  // tslint:disable-next-line:switch-default
+  switch (arg) {
+    case "--basePath":
+      basePath = resolve(argv[i + 1]);
+      break;
+    case "--buildPath":
+      buildPath = resolve(argv[i + 1]);
+      break;
+    case "--outFile":
+      outFile = resolve(argv[i + 1]);
+      break;
+    case "--silent":
+      silent = true;
+      break;
+  }
+});
```

Comment:
I'd rather not have top level scripts here. Can you wrap this all in a main function, and pass `process.argv` to it?